### DISCUSSION TITLE
Seeking for a approach to prevent repeated folder mounting in pyodide webworker.js

### DISCUSSION THREAD

**Discussion by jiyue1214** [pyodide/pyodide#4663] [Q&A] [answered]
# Issue Description:

Hi, all. I am new to using Pyodide and JavaScript. I have three buttons in my HTML script, each triggering a different Python script to process the same input file. I'm using a single webworker.js to handle these scripts and here is the script. 

in consumer.js
```
let context = {
        dirHandle: dirHandle,
        inputFileName: inputFileHandle.name,
        folder: folder,
};

const { results, error } = await asyncRun(format_input, context); // change the python script here for each button
```

in webworker.js
```
self.onmessage = async (event) => {
    var startTime = performance.now()
    await pyodideReadyPromise;
    const { id, python, ...context } = event.data;
    for (const key of Object.keys(context)) {
      self[key] = context[key];
    }
    try {
      console.log(self.folder);
      await self.pyodide.loadPackagesFromImports(python);
      // mount local directory
      const nativefs = await self.pyodide.mountNativeFS(self.folder, self.dirHandle);
      let results = await self.pyodide.runPythonAsync(python);
      await nativefs.syncfs();
      self.postMessage({ results, id });
    } catch (error) {
      self.postMessage({ error: error.message, id });
    }
  };
```
However, I'm encountering an issue where I have to mount the same directory within the webworker.js each time it's launched (three times for three buttons), and I must change the folder name to avoid an FS (file system) error.

I attempted to address this by using pyodide.mountNativeFS('/data', dirHandle) in consumer.js and transferring nativefs as part of the context. Unfortunately, this resulted in a "async()=>new Promise((n,a)=>Module.FS.syncfs(!1,n)) could not be cloned." error.

# Objective:

I'm seeking a better solution to avoid mounting the same folder in each web worker instance.

Any insights or guidance on how to achieve this more efficiently would be greatly appreciated.
Thank you for your assistance!

**Comment by ryanking13** [comment, marked as answer]
I think you can mount the filesystem once on initialization, and keep reusing them after that, is there some reason for mounting it every time?

**Reply by jiyue1214** [reply]
Thank you for your response, @ryanking13. You were correct, the issue was caused by not setting nativefs as a global variable, so that the new webworkers cannot read it 😅 (My JS level is at the point where to greet the console by saying 'Hello World'.").  I changed it and it works now. Thank you again.

```
if (! self.fsmounted){
        self.nativefs = await self.pyodide.mountNativeFS("/data", self.dirHandle);
        self.fsmounted = true;
      }
```
